kedro_vertexai/config.py

```diff
@@ -48,150 +62,66 @@
-            return False
-
-
-class VertexAiNetworkingConfig(Config):
-    @property
-    def vpc(self):
-        return self._get_or_default("vpc", None)
-
-    @property
-    def host_aliases(self):
-        aliases = self._get_or_default("host_aliases", [])
-        return {alias["ip"]: alias["hostnames"] for alias in aliases}
-
-
-class NodeResources(Config):
-    def is_set_for(self, node_name):
-        return self.get_for(node_name) != {}
-
-    def get_for(self, node_name):
-        defaults = self._get_or_default("__default__", {})
-        node_specific = self._get_or_default(node_name, {})
-        return {**defaults, **node_specific}
-
-
-class RunConfig(Config):
-    @property
-    def image(self):
-        return self._get_or_fail("image")
-
-    @property
-    def image_pull_policy(self):
-        return self._get_or_default("image_pull_policy", "IfNotPresent")
-
-    @property
-    def root(self):
-        return self._get_or_fail("root")
-
-    @property
-    def experiment_name(self):
-        return self._get_or_fail("experiment_name")
-
-    @property
-    def scheduled_run_name(self):
-        return self._get_or_default(
-            "scheduled_run_name", self._get_or_fail("experiment_name")
-        )
-
-    @property
-    def description(self):
-        return self._get_or_default("description", None)
-
-    @property
-    def resources(self):
-        return NodeResources(self._get_or_default("resources", {}))
-
-    @property
-    def store_kedro_outputs_as_kfp_artifacts(self):
-        return bool(
-            self._get_or_default("store_kedro_outputs_as_kfp_artifacts", True)
-        )
-
-    @property
-    def max_cache_staleness(self):
-        return str(self._get_or_default("max_cache_staleness", None))
-
-    @property
-    def ttl(self):
-        return int(self._get_or_default("ttl", 3600 * 24 * 7))
-
-    @property
-    def on_exit_pipeline(self):
-        return self._get_or_default("on_exit_pipeline", None)
-
-    @property
-    def vertex_ai_networking(self):
-        return VertexAiNetworkingConfig(
-            self._get_or_default("vertex_ai_networking", {})
-        )
-
-    @property
-    def node_merge_strategy(self):
-        strategy = str(self._get_or_default("node_merge_strategy", "none"))
-        if strategy not in ["none", "full"]:
-            raise ValueError(
-                f"Invalid {self._get_prefix()}node_merge_strategy: {strategy}"
+class NetworkConfig(BaseModel):
+    vpc: Optional[str]
+    host_aliases: Optional[List[HostAliasConfig]] = []
+
+
+class RunConfig(BaseModel):
+    image: str
+    image_pull_policy: Optional[str] = "IfNotPresent"
+    root: Optional[str]
+    description: Optional[str]
+    experiment_name: str
+    scheduled_run_name: Optional[str]
+    network: Optional[NetworkConfig] = NetworkConfig()
+    ttl: int = 3600 * 24 * 7
+    resources: Optional[Dict[str, ResourcesConfig]] = dict(
+        __default__=ResourcesConfig(cpu="500m", memory="1024Mi")
```

Comment:
What about using `from collections import defaultdict` with 
`resources: Optional[Dict[str, ResourcesConfig]] = defaultdict(lambda: ResourcesConfig(cpu="500m", memory="1024Mi"))`

it would also simplify (or even make not needed) the `resources_for` function.